src/net_processing.cpp

```diff
@@ -2902,10 +2910,14 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             // disk-space attacks), but this should be safe due to the
             // protections in the compact block handler -- see related comment
             // in compact block optimistic reconstruction handling.
-            ProcessNewBlock(chainparams, pblock, /*fForceProcessing=*/true, &fNewBlock);
-            if (fNewBlock) {
+            CValidationState dos_state;
+            ProcessNewBlock(chainparams, pblock, dos_state, /*fForceProcessing=*/true, &fNewBlock);
```

Comment:
Does it make sense to have a `ProcessAndNotifyNewBlock` to wrap this repeated code?

Reply:
I'd prefer to just wait and clean it up in a later PR, given I want to move towards doing this in the background anyway.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/16279#discussion_r297034762

In commit "Return the AcceptBlock CValidationState directly in ProcessNewBlock" (dd07ec726c5ade950435876d39741d52a33471ce)

> I'd prefer to just wait and clean it up in a later PR, given I want to move towards doing this in the background anyway

Seems to me deduplicating this code now would only simplify the later PR.

Reply:
Right, but in the full branch I first clean up some cruft, and then merge the now-identical code instead of the currently-partially-identical code. Easier to just leave it as-is.